core/src/main/scala/feral/IOSetup.scala

```diff
@@ -20,24 +20,36 @@ import cats.effect.unsafe.IORuntime
 import cats.effect.kernel.Resource
 import cats.effect.IO
 import cats.effect.kernel.Deferred
+import cats.effect.std.Dispatcher
 
 private[feral] trait IOSetup {
+  private def runtime =
+    IORuntime.global
 
-  protected def runtime: IORuntime = IORuntime.global
+  protected def dispatcher: Dispatcher[IO] =
```

Comment:
Meanwhile, this should probably be `final`. I am not sure in what case it should be overriden.